lib/Compiler.js

```diff
@@ -86,64 +87,108 @@ function compileSolc(embark, contractFiles, cb) {
   const outputDir = embark.config.buildDir + embark.config.contractDirectories[0];
   const solcConfig = embark.config.embarkConfig.options.solc;
 
-  const solc = shelljs.which('solc'); 
+  const solc = shelljs.which('solc');
   if (!solc) {
     logger.error('solc is not installed on your machine');
     logger.info('You can install it by following the instructions on: http://solidity.readthedocs.io/en/latest/installing-solidity.html');
     return cb('Compiler not installed');
   }
-  
+
   logger.info("compiling solidity contracts with command line solc...");
 
-  const allowedDirectories = contractFiles.map((contractFile) => path.dirname(contractFile.path))
-                                          .filter((x, i, a) => a.indexOf(x) === i);
-  
-  let compiled_object = {};
-  async.each(contractFiles,
-    function (file, callback) {
-      compileSolcContract(logger, file, allowedDirectories, solcConfig, (err, compileString) => {
-        if (err) {
-          return callback(err);
+  const allowedDirectories = contractFiles.map((contractFile) => path.join(process.cwd(), path.dirname(contractFile.path) + '/'))
+    .filter((x, i, a) => a.indexOf(x) === i);
+  // Add default contract paths
+  allowedDirectories.push(
+    path.join(process.cwd(), 'node_modules/'),
+    path.join(process.cwd(), '.embark/contracts/')
+  );
+
+  const remappings = []; // Will get populated by compilations
+  // Get content and remappings
+  async.each(contractFiles, (file, eachCb) => {
+    file.content(content => {
+      file.content = content;
```

Comment:
I am not sure to understand those 2 lines.
Why do you assign it? you have it already?

Reply:
Oh shit. Well, the goal was to add a new attribute to the `file` for the content so we don't have to parse the file with `content()` again, but never realized I just freaking removed the functio. Opps